feat(eslint-plugin-fiori-tools): add DataFieldForIntentBasedNavigation XML annotation rule (#4261)
SAP/open-ux-tools: .changeset/wacky-birds-fall.md

---
'@sap-ux/eslint-plugin-fiori-tools': minor
---

> [rule] Add rule to check that intent based navigation data fields ar enot used.

Reviewer: ```suggestion
[rule] Add rule to check that intent-based navigation data fields are not used.
```
